JShellAPI/src/main/java/org/togetherjava/jshellapi/service/DockerService.java

```diff
@@ -131,11 +243,24 @@ public void onNext(Frame object) {
                     }
                 }
             });
-
-        client.startContainerCmd(containerId).exec();
         return pipeIn;
     }
 
+    /**
+     * Checks if the Docker container with the given ID is currently running.
+     *
+     * @param containerId the ID of the container to check
+     * @return true if the container is running, false otherwise
+     */
+    public boolean isContainerRunning(String containerId) {
+        InspectContainerResponse containerResponse = client.inspectContainerCmd(containerId).exec();
+        return Boolean.TRUE.equals(containerResponse.getState().getRunning());
+    }
+
+    private String cachedContainerName() {
```

Comment:
Name is misleading since it is not a getter and it isn't pure. Rename it to newCachedContainerName.